i18n-helpers/src/bin/mdbook-gettext.rs

```diff
@@ -285,21 +285,21 @@ mod tests {
 
     #[test]
     fn test_translate_code_block() {
-        let catalog = create_catalog(&[(
-            "```rust,editable\n\
-             fn foo() {\n\n    let x = \"hello\";\n\n}\n\
-             ```",
-            "```rust,editable\n\
-             fn FOO() {\n\n    let X = \"guten tag\";\n\n}\n\
-             ```",
-        )]);
+        let catalog = create_catalog(&[
+            ("\"hello\"", "\"guten tag\""),
+            ("// line comment\n", "// linie kommentar\n"),
```

Comment:
This is a small thing, but would it be possible to trim away the final whitespace from one or more line comments?

It looks simpler in the PO file afterwards and it's simpler for the translators to deal with. It would also be consistent with how we turn `# Foo` into just `"Foo"` instead of `"Foo\n"`.

Reply:
I agree trimming the final whitespace (and indent spaces) is better for the translators, but I couldn't find an appropriate way for it.

If the linebreak is moved from `Group::Translate` to `Group::Skip`, I thought the continuous line comments are splitted into multiple messages. This causes translation difficulty because multi-lined paragraph can't be translated line by line for the differences of word order and length between languages.

Do you have any idea?

Reply:
I'll let @dyoo look into the details of the parsing — I think you're doing the right thing by putting multiple consecutive line comments into the same message.

Reply:
No immediate idea here so let's resolve this for now. I would love to land this and then we'll try it out for a bit before releasing it.